code/game/turfs/floors/map_floors.dm

```diff
@@ -0,0 +1,578 @@
+	name = "cave"
+	icon_state = "outer_cave_to_inner1"
+
+/turf/open/desertdam/cave/inner_cave_floor //inner cave floor
+	name = "cave"
+	icon_state = "inner_cave_1"
+
+//River
+/turf/open/desertdam/river
+	icon_state = "shallow_water_clean"
+
+/turf/open/desertdam/river/clean/shallow //shallow water
+	name = "river"
+	icon_state = "shallow_water_clean"
+
+/turf/open/desertdam/river/clean/shallow_edge //shallow water transition to deep
+	name = "river"
+	icon_state = "shallow_to_deep_clean_water1"
+
+/turf/open/desertdam/river/clean/deep_water_clean //deep water
+	name = "river"
+	icon_state = "deep_water_clean"
+
+/turf/open/desertdam/river/clean/shallow_water_desert_coast //shallow water coast
+	name = "river"
+	icon_state = "shallow_water_desert_coast1"
+
+/turf/open/desertdam/river/clean/shallow_water_desert_waterway //desert floor waterway
+	name = "river"
+	icon_state = "desert_waterway1"
+
+/turf/open/desertdam/river/clean/shallow_water_cave_coast //shallow water cave coast
+	name = "river"
+	icon_state = "shallow_water_cave_coast1"
+
+/turf/open/desertdam/river/clean/shallow_water_cave_waterway //cave floor waterway
+	name = "river"
+	icon_state = "shallow_water_cave_waterway1"
+
+//TOXIC
+/turf/open/desertdam/river/toxic
+	icon_state = "shallow_water_toxic"
+
+/turf/open/desertdam/river/toxic/shallow_water_toxic //shallow water
+	name = "river"
+	icon_state = "shallow_water_toxic"
+
+/turf/open/desertdam/river/toxic/shallow_edge_toxic //shallow water transition to deep
+	name = "river"
+	icon_state = "shallow_to_deep_toxic_water1"
+
+/turf/open/desertdam/river/toxic/deep_water_toxic //deep water
+	name = "river"
+	icon_state = "deep_water_toxic"
+
+/turf/open/desertdam/river/toxic/shallow_water_desert_coast_toxic //shallow water coast
+	name = "river"
+	icon_state = "shallow_water_desert_coast_toxic1"
+
+/turf/open/desertdam/river/toxic/shallow_water_desert_waterway_toxic //desert floor waterway
+	name = "river"
+	icon_state = "desert_waterway_toxic1"
+
+/turf/open/desertdam/river/toxic/shallow_water_cave_coast_toxic //shallow water cave coast
+	name = "river"
+	icon_state = "shallow_water_cave_coast_toxic1"
+
+/turf/open/desertdam/river/toxic/shallow_water_cave_waterway_toxic //cave floor waterway
+	name = "river"
+	icon_state = "shallow_water_cave_waterway_toxic1"
+
+
+// Mars grounds
+/turf/open/mars
+	name = "sand"
+	icon = 'icons/turf/bigred.dmi'
+	icon_state = "mars_sand_1"
+	is_groundmap_turf = TRUE
+
+
+/turf/open/mars_cave
+	name = "cave"
+	icon = 'icons/turf/bigred.dmi'
+	icon_state = "mars_cave_1"
+
+
+/turf/open/mars_cave/New()
+	..()
+	spawn(10)
+		var/r = rand(0, 2)
+
+		if (r == 0 && icon_state == "mars_cave_2")
+			icon_state = "mars_cave_3"
+
+/turf/open/mars_dirt
+	name = "dirt"
+	icon = 'icons/turf/bigred.dmi'
+	icon_state = "mars_dirt_1"
+
+/turf/open/mars_dirt/New()
+	..()
+	spawn(10)
+		if (rand(0, 32) && icon_state == "mars_dirt_4")

```

Comment:
This seems a little unintuitive. One chance in 33 not to change the icon to something else? Without an early return so then it checks again.

Why not a `rand` followed by a `switch` or a `prob` followed by a `pick`?

Reply:
probably for chance thing, instead of picking turf